src/legacy/status_im/browser/core.cljs

```diff
@@ -658,12 +659,11 @@
                           stored-bookmarks)]
     {:db (assoc db :bookmarks/bookmarks bookmarks)}))
 
-(rf/defn initialize-browser
-  [_]
-  {:json-rpc/call
-   [{:method     "wakuext_getBrowsers"
-     :on-success #(re-frame/dispatch [::initialize-browsers %])}
-    {:method     "browsers_getBookmarks"
-     :on-success #(re-frame/dispatch [::initialize-bookmarks %])}
-    {:method     "permissions_getDappPermissions"
-     :on-success #(re-frame/dispatch [::initialize-dapp-permissions %])}]})
+(re-frame/reg-fx :browser/initialize-browser
+ (fn []
+   (json-rpc/call {:method     "wakuext_getBrowsers"
```

Comment:
could you please elaborate on this change? im not sure i understand what's the point of this, why don't use effect?

Reply:
ok i think i understand now, and it's kinda fine

Reply:
Wait, but that's exactly what's happening here — this function is becoming an effect registered with `reg-fx` so that you are able to use it in `:fx` list in `:profile.login/get-chats-callback`.